src/modules/accounting/accounting.py

```diff
@@ -72,7 +73,7 @@ def refresh_contracts(self):
     def execute_module(self, last_finalized_blockstamp: BlockStamp) -> ModuleExecuteDelay:
         report_blockstamp = self.get_blockstamp_for_report(last_finalized_blockstamp)
 
-        if not report_blockstamp:
+        if not report_blockstamp or not self._check_compatability(report_blockstamp):
```

Comment:
Previous version was better
```suggestion
        if not report_blockstamp(report_blockstamp):
            return ModuleExecuteDelay.NEXT_FINALIZED_EPOCH
        
        if not self._check_compatability(report_blockstamp):
            return ModuleExecuteDelay.NEXT_FINALIZED_EPOCH
```